lib/p2p/packets/Packet.ts

```diff
@@ -104,10 +99,10 @@ abstract class Packet<T = any> implements PacketInterface {
   };
 
   /**
-   * Calculating the packet checksum using its JSON representation hash first 4 bytes.
+   * Calculating the packet checksum using its binary representation hash first 4 bytes.
    */
-  public checksum = (): number => {
-    return createHash('sha256').update(this.toJSON()).digest().readUInt32LE(0);
+  public checksum = (bytes: Uint8Array): number => {
```

Comment:
This is no longer using `this`, it's just a plain function that accepts a parameter. As such I'm thinking it might make sense to move this to a utils function.

Either that or it might make sense to remove the `bytes` parameter and just  do `.update(this.serialize())` to get the `UintArray` for this packet. Whichever makes more sense to you.

Reply:
> This is no longer using `this`, it's just a plain function that accepts a parameter. As such I'm thinking it might make sense to move this to a utils function.

Good idea.

> Either that or it might make sense to remove the `bytes` parameter and just  do `.update(this.serialize())` to get the `UintArray` for this packet. Whichever makes more sense to you.

I didn't want to trigger `.update(this.serialize())` because in both use-cases we already have it ready, so i'd rather not re-generate it.